p2p/conn/connection.go

```diff
@@ -503,6 +542,17 @@ FOR_LOOP:
 	}
 }
 
+// not goroutine-safe
+func (c *MConnection) stopPongTimer() {
+	if c.pongTimer != nil {
+		if !c.pongTimer.Stop() {
+			<-c.pongTimer.C
+		}
+		drain(c.pongTimeoutCh)
```

Comment:
I am fairly sure that we are better off avoiding draining functions.
If you're not convinced, I could try to put more thought into it, but I suspect that draining is not necessary, and consistently avoiding it is better.

Here, we can keep pongTimeoutCh capacity 0, and select on line 342.

Reply:
It's OK to leave things in channels, as long as nothing blocks on it.  We can drain defensively, but lets instead make sending to pongTimeoutCh nonblocking, and not drain.

Reply:
This is done in https://github.com/tendermint/tendermint/pull/1209, but is it really ok if we're always ignoring those timeouts ?